### ISSUE TITLE
About the results on model "capsuleEM2"

### ISSUE THREAD

**Issue by Wowoho** [jhui/machine_learning#1]
The results of on model "capsuleEM2" seems not correct.
The loss can't convergence, and the acc is always 0.1.

**Comment by ussaema** [comment]
same issue here!
any solution?

**Comment by msseibel** [comment]
The issue still exists. 
It seems like the output is always dominated by one class (like the output for 80% of the samples in one batch is the same).
e.g.
acc: 0.099999994
output:
 [3 3 3 3 3 3 3 6 4 3 3 4 3 3 4 3 3 3 3 3 3 3 3 4 3 3 3 4 3 3 3 3 3 3 4 4 3 3 3 3 3 3 3 3 3 4 3 3 3 3]

Can you add how to fix it, or how long it took until the acc increased?

**Comment by ussaema** [comment]
Check https://github.com/ussaema/Vector_Matrix_CapsuleGAN
It is in a GAN framework but you can use the part of the EM routing.

**Comment by msseibel** [comment]
They are training if you apply regularisation like Batchnorm or Dropout. You can also try to reduce the number of layers. Further keep in mind that a capsule layer with `num_routing_iterations` is comparable to having `num_routing_iterations` layers with the same weights. 
Maybe one can check if vanishing gradient is a problem or propose (and test) a different clustering method.
